tests/functionality.spec.js

```diff
@@ -81,6 +82,76 @@ describe('Test dependency sync', function () {
             'elm-community/spec-2': '1.0.0 <= v <= 1.0.0'
         });
     });
+
+    it('Should only change versions for common dependencies with --skip-missing', function () {
+
+        syncVersions(TOP_PATH, SPEC_PATH, true, false, false, true, undefined, undefined);
+
+        var specPkg = JSON.parse(fs.readFileSync(SPEC_PATH));
+
+        expect(specPkg.dependencies).to.be.deep.equal({
+            'elm-community/top-1': '1.0.0 <= v <= 1.0.0',
+            'elm-community/top-2': '1.0.0 <= v <= 1.0.0',
+            'elm-community/spec-1': '1.0.0 <= v <= 1.0.0',
+            'elm-community/spec-2': '1.0.0 <= v <= 1.0.0'
+        });
+
+    });
+
+    it('Should sync from dependencies to devDependencies', function () {
+
+        syncVersions(TOP_PATH, SPEC_PATH, true, false, false, false, 'dependencies', 'devDependencies');
+
+        var specPkg = JSON.parse(fs.readFileSync(SPEC_PATH));
+
+        expect(specPkg['devDependencies']).to.be.deep.equal({
+            'elm-community/top-1': '1.0.0 <= v <= 1.0.0',
+            'elm-community/top-2': '1.0.0 <= v <= 1.0.0',
+            'elm-community/top-3': '1.0.0 <= v <= 1.0.0'
+        });
+
+    });
+
+    it('Should sync from devDependencies to dependencies', function () {
+        var topLevelPkg = {
```

Comment:
This duplication is not beautiful but seemed the simplest way to test this without writing too many wrappers.